### PR TITLE
[core] Standardize weapon damage/wrank getters

### PR DESCRIPTION
**Pull request by WinterSolstice8**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

mob weapon damage is fetched correctly on autos (it was fine from lua auto attacks and any damage calculated in lua for the most part...)
wrank modifiers are applied correctly (more correctly?): see https://www.bg-wiki.com/ffxi/Weapon_Rank

player weapon damage is applied consistently (h2h was the edge case there)

reduces spaghetti
image

## Steps to test these changes

get slapped by tiamat and vrtra 
image
image

### CODE REVIEW

File: src/map/entities/battleentity.cpp [hunk 1]
```diff
@@ -585,6 +586,15 @@ int16 CBattleEntity::GetAmmoDelay()
 uint16 CBattleEntity::GetMainWeaponDmg()
 {
     TracyZoneScoped;
+
+    if (objtype == TYPE_MOB ||
```
Comment (UmeboshiXI) [inline src/map/entities/battleentity.cpp:590]: CLuaBaseEntity::getWeaponDmg() has this check logic as well. Not sure if you would want to leave it or just have the condition in GetMainWeaponDmg() handle the check.

uint16 CLuaBaseEntity::getWeaponDmg()
{
    if (m_PBaseEntity->objtype == TYPE_NPC)
    {
        ShowWarning("Invalid Entity (NPC: %s) calling function.", m_PBaseEntity->getName());
        return 0;
    }

    uint16 weaponDamage = 0;

    // TODO: Determine if trusts and player fellows use mob or player damage formula
    if (m_PBaseEntity->objtype == TYPE_MOB ||
        (m_PBaseEntity->objtype == TYPE_PET &&
         static_cast<CPetEntity*>(m_PBaseEntity)->getPetType() != PET_TYPE::AUTOMATON))
    {
        auto* PMob   = static_cast<CMobEntity*>(m_PBaseEntity);
        weaponDamage = mobutils::GetWeaponDamage(PMob, SLOT_MAIN);
    }
    else
    {
        weaponDamage = static_cast<CBattleEntity*>(m_PBaseEntity)->GetMainWeaponDmg();
    }

    return weaponDamage;
}
```

**Review by Xaver-DaRed — APPROVED** [review]
LGTM
Address Umeboshi's comment and merge away if it's ready.

File: src/map/entities/battleentity.cpp (hunk 1 above)
Comment (WinterSolstice8) [reply src/map/entities/battleentity.cpp:590]: done
